app/src/main/java/com/example/marvelapp/presentation/adapter/CharacterAdapter.kt

```diff
@@ -9,29 +9,44 @@ import com.example.marvelapp.R
 import com.example.marvelapp.databinding.CharacterCardLayoutBinding
 import com.example.marvelapp.domain.entity.CharacterEntity
 
-class CharacterAdapter(private val character: List<CharacterEntity>) :
+interface ItemClicked {
+    fun characterClicked(characterId: Int)
+}
+
+class CharacterAdapter(
+    private val character: List<CharacterEntity>,
+    private val itemClickedListener: ItemClicked
+) :
     RecyclerView.Adapter<CharacterAdapter.ViewHolder>() {
+
     override fun onCreateViewHolder(parent: ViewGroup, viewType: Int) =
         ViewHolder(
             LayoutInflater.from(parent.context)
                 .inflate(R.layout.character_card_layout, parent, false)
         )
 
     override fun onBindViewHolder(holder: ViewHolder, position: Int) {
-        holder.bind(character[position])
+        val item = character[position]
+        holder.bind(item, itemClickedListener)
     }
 
     override fun getItemCount() = character.size
 
-    class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
+    class ViewHolder(itemView: View) :
+        RecyclerView.ViewHolder(itemView) {
         private val binding = CharacterCardLayoutBinding.bind(itemView)
 
-        fun bind(characterItem: CharacterEntity) {
+        fun bind(characterItem: CharacterEntity, itemClickedListener: ItemClicked) {
             binding.apply {
                 heroName.text = characterItem.name
                 Glide.with(itemView.context)
                     .load("${characterItem.thumbnail.path}.${characterItem.thumbnail.extension}")
                     .into(imageThumbnail)
+                cardviewCharacterFrag.setOnClickListener {
+                    itemClickedListener.characterClicked(
+                        characterItem.id
+                    )
+                }
```

Comment:
why do you declare an interface to use the on click listener?

Reply:
Because currently it allows me to make the fragment dialog work, and in the next functionality I will also use it